dsc_lib/src/configure/config_doc.rs

```diff
@@ -71,6 +71,7 @@ pub struct Resource {
     /// A friendly name for the resource instance
     pub name: String, // friendly unique instance name
     #[serde(rename = "dependsOn", skip_serializing_if = "Option::is_none")]
+    #[schemars(regex(pattern = r"^\[resourceId\('[a-zA-Z0-9\.]+/[a-zA-Z0-9]+','[a-zA-Z0-9 ]+'\)]$"))]
```

Comment:
Can this allow for an optional space between the type name and the instance name? Does it _have_ to use single quotes?

Thinking about examples like:

```yaml
resources:
- name: current user registry
  type: Microsoft.Windows/Registry
  properties:
    keyPath: HKCU\example
    _ensure: Present
  dependsOn:
  # Current implementation
  - "[resourceId('DSC/AssertionGroup','my assertions')]"
  # With outer quotes as single quotes
  - '[resourceId("DSC/AssertionGroup","my assertions")]'
  # With a space between the names
  - "[resourceId('DSC/AssertionGroup', 'my assertions')]"
```

The regex for the type name also doesn't match our current regex - I'm okay with adjusting either of them:

https://github.com/PowerShell/DSC/blob/5a1f8b582edfbe8a040195baa7ff4bdb7231addd/schemas/src/definitions/resourceType.yaml#L3-L17

We also don't have a regex currently applied to instance names, we allow arbitrary strings. Should that be more limited?

Reply:
Best I can tell from ARM, whitespace is allowed.  I'll adjust the regex.

Reply:
With the updated whitespace, dangling questions are:

1. Can the regex also support using double quotes instead of single quotes?
2. Can the instance names be arbitrary strings or are they limited to alphanumeric characters and spaces? If they're limited, I can adjust the docs and schemas instead of allowing arbitrary strings. Right now, this regex makes the following instance names invalid:
  ```yaml
  - name: Security.PublicAccess.Disable
  - name: Security/PublicAccess/Disable
  - name: Security - Disable public access
  - name: 'Security: Disable public access'
  ```

Reply:
@michaeltlombardi best I can tell looking at the ARM docs it's single quote only because it's in JSON which would get wrapped by double quotes.  Don't see any specific rules on characters, but the examples are all alphanumeric with spaces allowed, so I think that's ok unless you can find something more specific to what ARM allows

Reply:
We can update this as a separate PR as needed